### PR TITLE
Fix issue with messaging mode not working on iOS

### PR DESCRIPTION
**Pull request by RasimKanca**
The largest change in this request is the rewrite of the `init()` function, which supports multiple options through the options parameter. Part of the rewrite was also to solve issues with the "messaging mode" on iOS. This option has been renamed to `isTestDevice` and now only has to be set on initialisation. I have also added a `getDeviceIdentifier()` function that allows the device identifier to be retrieved and have done some cleanup.

### CODE REVIEW

**Comment by trinisofttechnologies** [comment]
@RasimKanca 

I am reviewing this issue, and try to merge it in such a way, that it doesn't break existing customers.

Thanks

File: src/ios/CountlyCordova.m [hunk 1]
```diff
@@ -15,49 +15,51 @@ - (void)sendToken;
...
-    
-    if(command.arguments.count == 3){
-        deviceID = [command.arguments objectAtIndex:2];
-        config.deviceID = deviceID;
+
+    if (command.arguments.count == 3) {
+        NSDictionary *options = [command.arguments objectAtIndex:2];
+
+        if ([options objectForKey:@"isTestDevice"]) {
+            // isTestDevice only works when the Push Notification feature is enabled.
```
Comment (iartem) [inline src/ios/CountlyCordova.m:37]: @RasimKanca `CLYPushNotifications` is required for any case when push is needed, not really related to `isTestDevice`

**Comment by RasimKanca** [comment]
As most of what I proposed has already been added to the plugin in one way or another, I am going to go ahead and close this.

File: src/ios/CountlyCordova.m (hunk 1 above)
Comment (RasimKanca) [reply src/ios/CountlyCordova.m:37]: You are absolutely right, but here I am assuming that people that want to use push notifications pass `isTestDevice`.

**Pull request closed without merging**